lightning/src/ln/onion_route_tests.rs

```diff
@@ -594,6 +593,93 @@ fn test_onion_failure() {
+        &nodes, 1, 2, InitFeatures::known(), InitFeatures::known(),
+    );
+
+    // A test payment should succeed as the HTLC relay paramters have not been changed yet.
+    const PAYMENT_AMT: u64 = 40000;
+    send_payment(&nodes[0], &vec!(&nodes[1], &nodes[2])[..], PAYMENT_AMT);
+
+    // Closure to update and retrieve the latest ChannelUpdate.
+    let update_and_get_channel_update =
+		|updates: &[HTLCRelayPolicyUpdate], expect_new_update: bool,
+		prev_update: Option<&msgs::ChannelUpdate>| -> Option<msgs::ChannelUpdate> {
+			nodes[1].node.update_htlc_relay_policy(&[channel_to_update.2], updates).unwrap();
+			let events = nodes[1].node.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), expect_new_update as usize);
+			if !expect_new_update {
+				return None;
+			}
+			let new_update = match &events[0] {
+				MessageSendEvent::BroadcastChannelUpdate { msg, .. } => msg.clone(),
+				_ => panic!("expected BroadcastChannelUpdate event"),
+			};
+			if prev_update.is_some() {
+				assert!(new_update.contents.timestamp > prev_update.unwrap().contents.timestamp)
+			}
+			Some(new_update)
+		};
+
+	// We'll be attempting to route payments using the default ChannelUpdate for channels. This will
+	// lead to onion failures at the first hop once we update the HTLC relay parameters for the
+	// second hop.
+	let (route, payment_hash, _, payment_secret) = get_route_and_payment_hash!(
+		nodes[0], nodes[2], PAYMENT_AMT
+	);
+	let expect_onion_failure = |name: &str, error_code: u16, channel_update: &msgs::ChannelUpdate| {
+		let short_channel_id = channel_to_update.0.contents.short_channel_id;
+		let network_update = NetworkUpdate::ChannelUpdateMessage { msg: channel_update.clone() };
+		run_onion_failure_test(
+			name, 0, &nodes, &route, &payment_hash, &payment_secret, |_| {}, || {}, true,
+			Some(error_code), Some(network_update), Some(short_channel_id),
+		);
+	};
+
+	// Updates to cltv_expiry_delta below MIN_CLTV_EXPIRY_DELTA should fail with APIMisuseError.
+	let updates = &[HTLCRelayPolicyUpdate::CltvExpiryDelta(Some(MIN_CLTV_EXPIRY_DELTA - 1))];
+	match nodes[1].node.update_htlc_relay_policy(&[channel_to_update.2], updates) {
+		Err(APIError::APIMisuseError{ .. }) => {},
+		_ => panic!("unexpected result applying invalid cltv_expiry_delta"),
+	}
+
+	// Increase the base fee which should trigger a new ChannelUpdate.
+	let updates = &[HTLCRelayPolicyUpdate::ForwardingFeeBaseMsat(Some(u32::MAX))];
+	let msg = update_and_get_channel_update(updates, true, None).unwrap();
+	expect_onion_failure("fee_insufficient", UPDATE|12, &msg);
+
+	// Redundant updates should not trigger a new ChannelUpdate.
+	assert!(update_and_get_channel_update(updates, false, None).is_none());
+
+	// Reset the base fee to the default and increase the proportional fee which should trigger a
+	// new ChannelUpdate.
+	let updates = &[
+		HTLCRelayPolicyUpdate::ForwardingFeeBaseMsat(None),
+		HTLCRelayPolicyUpdate::CltvExpiryDelta(Some(u16::MAX)),
```

Comment:
`MAX`/`MIN` constants have only been defined since Rust 1.43.0. Since MSRV is 1.41.1, you probably still need to use `max_value()`, even though it will be deprecated in the future.

```suggestion
		HTLCRelayPolicyUpdate::CltvExpiryDelta(Some(u16::max_value())),
```